src/cdk-experimental/menu/background-click-service.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, Inject, InjectionToken, OnDestroy} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {fromEvent, Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+import {MenuStack} from './menu-stack';
+
+/**
+ * A function which decides whether the background click event should cause the menu stack to close
+ * out. It is provided alongside the MenuStack to the BackgroundClickService and is called on each
+ * background click event.
+ */
+export type CloseDecider = (target: Element | null) => boolean;
+
+/** Injection token for CloseDecider function. */
+export const CLOSE_DECIDER = new InjectionToken<CloseDecider>('cdk-menu-close-decider');
+
+@Injectable({providedIn: 'root'})
```

Comment:
This should have some documentation, especially since I'm a little unclear whether the client developer will call this directly, or whether it will only be internally invoked.

Reply:
@teflonwaffles done